src/bitcoind/interface.rs

```diff
@@ -647,28 +700,48 @@ impl BitcoinD {
     pub fn sync_unvaults(
         &self,
         unvault_utxos: &HashMap<OutPoint, UtxoInfo>,
+        db_tip: &BlockchainTip,
     ) -> Result<UnvaultsState, BitcoindError> {
-        // Since we don't need to care about new utxos the logic here is more
-        // straightforward than in sync_deposits.
-        //
-        // 1. Fetch the Unvault utxos from the watchonly wallet into a
-        //    (outpoint, confirmed) mapping
-        let unspent_list: HashMap<OutPoint, bool> = self
-            .list_unspent_unvaults(None)?
-            .iter()
-            .map(|utxo| (utxo.outpoint, utxo.confirmations > 0))
-            .collect();
-
-        // 2. Loop through all known Unvault utxos, check if some confirmed or
-        //    are missing (ie were spent)
         let (mut new_conf, mut new_spent) = (HashMap::new(), HashMap::new());
-        for (op, utxo_info) in unvault_utxos {
-            if let Some(confirmed) = unspent_list.get(op) {
-                if *confirmed && !utxo_info.is_confirmed {
-                    new_conf.insert(*op, utxo_info.clone());
+        let utxos = self.list_since_block(db_tip)?;
+
+        for utxo in utxos {
+            let is_confirmed = utxo.confirmations > 0;
+            if utxo.category == "receive"
+                && is_confirmed
+                && utxo.label == Some(UNVAULT_UTXOS_LABEL.to_string())
+            {
+                if let Some(UtxoInfo {
+                    is_confirmed: false,
+                    ..
+                }) = unvault_utxos.get(&utxo.outpoint)
+                {
+                    new_conf.insert(
+                        utxo.outpoint,
+                        UtxoInfo {
+                            txo: utxo.txo,
+                            is_confirmed,
+                        },
+                    );
+                }
+            } else if utxo.category == "send" {
+                // If we sent this utxo, then it's spent. If it was received before `db_tip` then it's
+                // in the cache already, otherwise we will see it and insert it
+                // We retrieve the transaction and mark all the inputs as spent
+                let tx_hex = self.get_wallet_transaction(&utxo.outpoint.txid)?.hex;
+                let tx = Vec::<u8>::from_hex(&tx_hex).expect("Must be valid hex");
+                let tx = Transaction::consensus_decode(tx.as_slice()).expect("Must be a valid tx");
+                for input in tx.input {
+                    if let Some(d) = unvault_utxos.get(&input.previous_output) {
+                        new_spent.insert(
+                            input.previous_output,
+                            UtxoInfo {
+                                txo: d.txo.clone(),
+                                is_confirmed,
+                            },
+                        );
+                    }
```

Comment:
I think this might be the cause of the timeouts you mentioned. Imagine a Spend transaction with 588 outputs (as we do in the not announceable spend test) and 20 inputs (same test). This would loop 11_760 times.. (and this is combinatorial, this can quickly get much bigger).
I think an approach checking if all our utxos are still unspent using `gettxout`, and then using `listsinceblock` to detect new outputs paying to us (`receive`) would be reasonable.
Even for the deposits, i think using the `send` results from `listsinceblock` is a bit "clumsy" and an approach coupling `gettxout` for detecting the spent utxos and `listsinceblock` for detecting the received ones should be prefered.